### PR TITLE
Split unit test and unit race into 2 components

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR splits the `unit_test` and `unit_race` workflows into 2 parts, one that runs the evalengine tests only, and the other that runs everything else. This has been done because the CI workflows started taking too long and necessicated splitting them up.

The default behaviour of `make unit_test` and `make unit_race` doesn't change. In order to run restricted set of tests like we do in the CI, `VTEVALENGINETEST` environment variable has to be set to `"1"` or `"-1"`.
## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by rohit-nayak-ps — APPROVED** [review]

File: .github/workflows/unit_race.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-name: unit_race
+name: unit_race_non_evalengine
```
Comment (harshit-gangal) [inline .github/workflows/unit_race.yml:1]: let's keep this unchanged

File: .github/workflows/unit_race.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ env:
 jobs:
 
   build:
-    name: Unit Test (Race)
+    name: Unit Test (Non_Evalengine_Race)
```
Comment (deepthi) [inline .github/workflows/unit_race.yml:17]: Along the lines of harshit's comment, it makes sense to revert this change as well.

File: .github/workflows/unit_test_non_evalengine_mysql57.yml [hunk 3]
```diff
@@ -0,0 +1,172 @@
+# DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
+
+name: Unit Test (non_evalengine_mysql57)
```
Comment (deepthi) [inline .github/workflows/unit_test_non_evalengine_mysql57.yml:3]:
```suggestion
name: Unit Test (mysql57)
```

File: .github/workflows/unit_test_non_evalengine_mysql57.yml [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+permissions: read-all
+
+env:
+  LAUNCHABLE_ORGANIZATION: "vitess"
+  LAUNCHABLE_WORKSPACE: "vitess-app"
+  GITHUB_PR_HEAD_SHA: "${{ github.event.pull_request.head.sha }}"
+
+jobs:
+  test:
+    name: Unit Test (non_evalengine_mysql57)
```
Comment (deepthi) [inline .github/workflows/unit_test_non_evalengine_mysql57.yml:18]:
```suggestion
    name: Unit Test (mysql57)
```

File: .github/workflows/unit_test_non_evalengine_mysql80.yml [hunk 5]
```diff
@@ -0,0 +1,169 @@
+# DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
+
+name: Unit Test (non_evalengine_mysql80)
```
Comment (deepthi) [inline .github/workflows/unit_test_non_evalengine_mysql80.yml:3]:
```suggestion
name: Unit Test (mysql80)
```

File: .github/workflows/unit_test_non_evalengine_mysql80.yml [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+permissions: read-all
+
+env:
+  LAUNCHABLE_ORGANIZATION: "vitess"
+  LAUNCHABLE_WORKSPACE: "vitess-app"
+  GITHUB_PR_HEAD_SHA: "${{ github.event.pull_request.head.sha }}"
+
+jobs:
+  test:
+    name: Unit Test (non_evalengine_mysql80)
```
Comment (deepthi) [inline .github/workflows/unit_test_non_evalengine_mysql80.yml:18]:
```suggestion
    name: Unit Test (mysql80)
```

Comment (deepthi) [reply .github/workflows/unit_test_non_evalengine_mysql80.yml:18]: I realize you need to change workflow gen to make these changes happen. It was easiest to suggest changes this way than to ci_workflow_gen.go

File: test/templates/unit_test.tpl [hunk 7]
```diff
@@ -170,6 +170,8 @@ jobs:
         export VTDATAROOT="/tmp/"
 
         export NOVTADMINBUILD=1
+        export VT_EVALENGINE_TEST="{{.Evalengine}}"
```
Comment (deepthi) [inline test/templates/unit_test.tpl:173]: nit: other env variables don't use `_` at all.

File: .github/workflows/unit_test_non_evalengine_mysql80.yml (hunk 6 above)
Comment (deepthi) [reply .github/workflows/unit_test_non_evalengine_mysql80.yml:18]: As a bonus, keeping workflow names same as before will mean that you don't have "3 expected workflows" on this PR which don't exist any more.

**Review by deepthi — APPROVED** [review]

File: .github/workflows/unit_race.yml [hunk 8]
```diff
@@ -130,6 +130,7 @@ jobs:
         # which musn't be more than 107 characters long.
         export VTDATAROOT="/tmp/"
         export NOVTADMINBUILD=1
+        export VTEVALENGINETEST="-1"
```
Comment (deepthi) [inline .github/workflows/unit_race.yml:133]: One last nit: -1 is a weird value for this env variable. Should have been 0 for false, and 1 for true.

Comment (GuptaManan100) [reply .github/workflows/unit_race.yml:133]: Okay, I'll change it 👍
